diff --git a/node_modules/@greenwood/plugin-import-jsx/src/index.js b/node_modules/@greenwood/plugin-import-jsx/src/index.js
index bc7e218..83b93bb 100644
--- a/node_modules/@greenwood/plugin-import-jsx/src/index.js
+++ b/node_modules/@greenwood/plugin-import-jsx/src/index.js
@@ -7,22 +7,21 @@ import escodegen from "escodegen";
 import { parseJsx } from "wc-compiler/src/jsx-loader.js";
 
 class ImportJsxResource {
-  constructor(compilation) {
+  constructor(compilation, options = {}) {
     this.compilation = compilation;
-    this.extensions = ["jsx"];
+    this.extensions = ["jsx", "tsx"];
     this.contentType = "text/javascript";
+    this.servePage = options.servePages ? 'dynamic' : null;
   }
 
   async shouldServe(url) {
     const { pathname, protocol } = url;
     const ext = pathname.split(".").pop();
 
-    return protocol === "file:" && ext === this.extensions[0];
+    return protocol === "file:" && this.extensions.includes(ext);
   }
 
   async serve(url) {
-    // refactor when WCC refactors
-    // https://github.com/ProjectEvergreen/wcc/issues/116
     const tree = parseJsx(url);
     const result = escodegen.generate(tree);
 
@@ -35,11 +34,11 @@ class ImportJsxResource {
 }
 
 /** @type {import('./types/index.d.ts').ImportJsxPlugin} */
-const greenwoodPluginImportJsx = () => [
+const greenwoodPluginImportJsx = (options = {}) => [
   {
     type: "resource",
     name: "plugin-import-jsx:resource",
-    provider: (compilation) => new ImportJsxResource(compilation),
+    provider: (compilation) => new ImportJsxResource(compilation, options),
   },
 ];
 
diff --git a/node_modules/@greenwood/plugin-import-jsx/src/types/index.d.ts b/node_modules/@greenwood/plugin-import-jsx/src/types/index.d.ts
index 0f42b24..8c02830 100644
--- a/node_modules/@greenwood/plugin-import-jsx/src/types/index.d.ts
+++ b/node_modules/@greenwood/plugin-import-jsx/src/types/index.d.ts
@@ -1,6 +1,10 @@
 import type { ResourcePlugin } from "@greenwood/cli";
 
-export type ImportJsxPlugin = () => [ResourcePlugin];
+type Options = {
+  servePages: boolean;
+}
+
+export type ImportJsxPlugin = (options?: Options) => [ResourcePlugin];
 
 declare module "@greenwood/plugin-import-jsx" {
   export const greenwoodPluginImportJsx: ImportJsxPlugin;
