diff --git a/node_modules/@greenwood/cli/src/types/ssr.d.ts b/node_modules/@greenwood/cli/src/types/ssr.d.ts
index 95e35e6..65c3574 100644
--- a/node_modules/@greenwood/cli/src/types/ssr.d.ts
+++ b/node_modules/@greenwood/cli/src/types/ssr.d.ts
@@ -37,7 +37,7 @@ export type StaticPath = { params: object };
 export type StaticParam = Record<string, unknown>;
 
 export type GetStaticPaths = () => Promise<StaticPath[]>;
-export type GetStaticParams = ({ params }) => Promise<StaticParam>;
+export type GetStaticParams = ({ params }: any) => Promise<StaticParam>;
 
 export type InferGetStaticParamsType<T> = T extends () => Promise<Array<{ params: infer P }>>
   ? P
