diff --git a/node_modules/wc-compiler/src/jsx.d.ts b/node_modules/wc-compiler/src/jsx.d.ts
index 160e5dc..8d36c4b 100644
--- a/node_modules/wc-compiler/src/jsx.d.ts
+++ b/node_modules/wc-compiler/src/jsx.d.ts
@@ -49,6 +49,17 @@ type IntrinsicElementsFromSVG = {
   [E in keyof SVGElementTagNameMap]: SVGElementAttributes<SVGElementTagNameMap[E]>;
 };
 
+type IntrinsicElementAttributes<E extends string> = (E extends keyof HTMLElementTagNameMap
+  ? ElementAttributes<HTMLElementTagNameMap[E]> &
+      (E extends 'button' | 'input' ? PopoverAttributes : {}) &
+      (E extends 'button' ? { tabindex?: number | string } : {})
+  : {}) &
+  (E extends keyof SVGElementTagNameMap ? SVGElementAttributes<SVGElementTagNameMap[E]> : {});
+
+type IntrinsicElementsMap = {
+  [E in keyof HTMLElementTagNameMap | keyof SVGElementTagNameMap]: IntrinsicElementAttributes<E>;
+};
+
 declare namespace JSX {
-  interface IntrinsicElements extends IntrinsicElementsFromDom, IntrinsicElementsFromSVG {}
+  interface IntrinsicElements extends IntrinsicElementsMap {}
 }
\ No newline at end of file
